samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -18,6 +18,7 @@
 
 #include "app_fp_adv.h"
 #include "app_ui.h"
+#include "app_dfu.h"
```

Comment:
To minimize the number of modules that have circular dependency, let's consider the following:

- remove the `dfu_adv_enable` function from here
- add the `enum app_fp_adv_mode app_fp_adv_mode_get(void)` API in the `app_fp_adv.h` header

Then, the fp_adv module doesn't need to be app_dfu aware :)

The app_dfu advertising provider can pick the correct way of using advertising/scanning data based on its internal state (DFU mode) and using the `app_fp_adv_mode_get` function.

Please let me know if you see any issue with this improvement.